src/Cookie/FileCookieJar.php

```diff
@@ -44,8 +49,10 @@ public function save($filename)
         $json = [];
         foreach ($this as $cookie) {
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
-                $json[] = $cookie->toArray();
+            if($cookie->getExpires() || $this->storeSessionCookies) {
```

Comment:
What do you think about refactoring this duplication of if statements out into a static method on CookieJar? Maybe the method could be `shouldPersist(SetCookie $cookie, $allowSessionCookies)` or something similar?